### PR TITLE
[CoreSpotlight] Add support for Xcode 14 beta 1,2 & 3

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

File: src/corespotlight.cs [hunk 1]
```diff
@@ -1105,4 +1117,123 @@ interface CSImportExtension : NSExtensionRequestHandling
...
+		[Export ("userQueryContextWithCurrentSuggestion:")]
+		CSUserQueryContext Create ([NullAllowed] CSSuggestion currentSuggestion);
+
+		[Export ("maxSuggestionCount")]
+		nint MaxSuggestionCount { get; set; }
+	}
+
+	[NoTV, Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+	[BaseType (typeof (CSUserQueryContext))]
+	interface CSTopHitQueryContext
```
Comment (dalexsoto) [inline src/corespotlight.cs:1160]: There is a create method here, I would Disable the default ctor here

```suggestion
	[DisableDefaultCtor]
	interface CSTopHitQueryContext
```

File: src/corespotlight.cs [hunk 2]
```diff
@@ -1105,4 +1117,123 @@ interface CSImportExtension : NSExtensionRequestHandling
...
+		[Export ("start")]
+		void Start ();
+
+		[Export ("cancel")]
+		void Cancel ();
+	}
+
+	[NoTV, Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+	[BaseType (typeof (CSSearchQueryContext))]
+	interface CSUserQueryContext
```
Comment (dalexsoto) [inline src/corespotlight.cs:1144]: Same, there is a create method here

```suggestion
	[DisableDefaultCtor]
	interface CSUserQueryContext
```

File: src/corespotlight.cs [hunk 3]
```diff
@@ -1105,4 +1117,123 @@ interface CSImportExtension : NSExtensionRequestHandling
...
+	interface CSTopHitQuery
+	{
+		[Export ("initWithUserQueryString:queryContext:")]
+		[DesignatedInitializer]
+		NativeHandle Constructor ([NullAllowed] string userQueryString, CSTopHitQueryContext queryContext);
+	}
+
+	[NoTV, Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+	[BaseType (typeof (NSObject))]
+	interface CSSuggestion : NSSecureCoding, NSCopying
```
Comment (dalexsoto) [inline src/corespotlight.cs:1186]: This seems to be used as a "Result" object and it is readonly

```suggestion
	[DisableDefaultCtor]
	interface CSSuggestion : NSSecureCoding, NSCopying
```

File: src/corespotlight.cs [hunk 4]
```diff
@@ -1105,4 +1117,123 @@ interface CSImportExtension : NSExtensionRequestHandling
...
+	}
+
+	[NoTV, Mac (10,13), iOS (16,0), MacCatalyst (16,0)]
+	[Native]
+	public enum CSSuggestionKind : long
+	{
+		None,
+		Custom,
+		Default,
+	}
```
Comment (dalexsoto) [inline src/corespotlight.cs:1237]: Nit: Not a fan of having these at the bottom of the file, we usually put them at the top but I won't die on a hill for this.

Comment (mandel-macaque) [reply src/corespotlight.cs:1240]: There are no enums in this file but this ones ;) I can move them, I'm not to bordered to be honest.

File: src/corespotlight.cs [hunk 5]
```diff
@@ -90,6 +90,13 @@ interface CSSearchableIndex {
 		[Export ("deleteAllSearchableItemsWithCompletionHandler:")]
 		[Async]
 		void DeleteAll ([NullAllowed] Action<NSError> completionHandler);
+
+		// from interface CSExternalProvider (CSSearchableIndex)
+
+		[Async (ResultTypeName="CSSearchableIndexBundleDataResult")]
+		[NoTV, iOS (16,0), Mac (13,0)]
```
Comment (tj-devel709) [inline src/corespotlight.cs:97]: Are you missing MacCatalyst annotations?

File: src/corespotlight.cs [hunk 6]
```diff
@@ -1105,4 +1117,126 @@ interface CSImportExtension : NSExtensionRequestHandling
...
+		[Static]
+		[Export ("topHitQueryContextWithCurrentSuggestion:")]
+		CSTopHitQueryContext Create ([NullAllowed] CSSuggestion currentSuggestion);
+
+		[Export ("maxItemCount")]
+		nint MaxItemCount { get; set; }
+	}
+
+	[NoTV, Mac (13,0), iOS (16,0), MacCatalyst (16,0)]
+	[BaseType (typeof(CSUserQuery))]
```
Comment (tj-devel709) [inline src/corespotlight.cs:1177]:
```suggestion
	[BaseType (typeof (CSUserQuery))]
```
nit

**Review by dalexsoto — APPROVED** [review]

**Pull request merged**